src/primitives/transaction.h

```diff
@@ -70,13 +70,29 @@ class CTxIn
     uint32_t nSequence;
     CScriptWitness scriptWitness; //!< Only serialized through CTransaction
 
-    /* Setting nSequence to this value for every input in a transaction
-     * disables nLockTime. */
+    /**
+     * Setting nSequence to this value for every input in a transaction
+     * disables nLockTime/IsFinalTx().
+     * It fails OP_CHECKLOCKTIMEVERIFY/CheckLockTime() for any input that has
+     * it set (BIP 65).
+     * It has SEQUENCE_LOCKTIME_DISABLE_FLAG set (BIP 68/112).
+     */
     static const uint32_t SEQUENCE_FINAL = 0xffffffff;
+    /**
+     * This is the maximum sequence number that enables both nLockTime and
+     * OP_CHECKLOCKTIMEVERIFY (BIP 65).
+     * It has SEQUENCE_LOCKTIME_DISABLE_FLAG set (BIP 68/112).
+     */
+    static const uint32_t MAX_SEQUENCE_NONFINAL{SEQUENCE_FINAL - 1};
 
     /* Below flags apply in the context of BIP 68*/
-    /* If this flag set, CTxIn::nSequence is NOT interpreted as a
-     * relative lock-time. */
+    /**
+     * If this flag is set, CTxIn::nSequence is NOT interpreted as a
+     * relative lock-time.
+     * It skips SequenceLocks() for any input that has it set (BIP 68).
+     * It fails OP_CHECKSEQUENCEVERIFY/CheckSequence() for any input that has
+     * it set (BIP 112).
```

Comment:
You are using the same language here as for BIP65 timelocks, whereas in this case it's "local" (if an input needs both satisfy a BIP112 timelock and its nSequence has the disable flag) while in the other it's "global" (if any input of the spending transaction has a BIP65 timelock and any other input has its nSequence set to SEQUENCE_FINAL).

I don't have a better suggestion. Just a nit that it *might* be confusing to someone only from these comments.

Reply:
Sorry, can you explain this a bit better? Both `OP_CHECKLOCKTIMEVERIFY` and `OP_CHECKSEQUENCEVERIFY` are evaluated in a local context of the input.

Sure, the whole transaction fails "globally", if any input is invalid, but this is true for both opcodes as well.

Reply:
Brainfart, sorry. I for some reason mistakenly assumed that `IsFinalTx()` returned `true` if any input's `nSequence` was set to `SEQUENCE_FINAL`... And of course didn't double check before posting this confusing comment.

Reply:
No worries. Going to resolve the thread for now.